### PR TITLE
Complete rewrite using html scraping

### PR DESCRIPTION
**Pull request by e-e-e**
Shifted from dead XML to html scraping

### CODE REVIEW

**Comment by e-e-e** [comment]
@henare is there a problem this this one? I ask not to pester, just to learn.
<3 Benjamin

**Comment by henare** [comment]
Ha, I'd just opened this to review it now :smile:

**Comment by e-e-e** [comment]
My first ruby - so any code tips and comments super welcome.

File: scraper.rb [hunk 1]
```diff
@@ -1,3 +1,55 @@
...
+
+def extract_from_p(contents, name)
+	contents.at('p:contains("'+name+'")').inner_text.split(name).last.strip
+end
+
+def scrape_page(page)
+	contents = page.search('div.span6')
+	record = {
+		"info_url" => page.uri.to_s,
+		"council_reference" => contents.search('h1').inner_text,
```
Comment (henare) [inline scraper.rb]: Council reference needs whitespace around it stripped:

selection_001

File: scraper.rb [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+		"comment_url" => page.uri.to_s,
+		"applicatant" => extract_from_p(contents,"Applicant:"),
+		"description" => contents.at('p:contains("Location:")').next_element.inner_text,
+		"address" => extract_from_p(contents,"Location:"),
+		"date_scraped" => Date.today.to_s,
+		"date_closing" => extract_from_p(contents,"Advertising closes ")
```
Comment (henare) [inline scraper.rb]: This string should be parsed into a date or at least a string in the "YYYY-MM-DD" format.

**Comment by henare** [comment]
I've added a couple of comments for things that need fixing. Once they're done this is ready for merge :shipit:

**Comment by henare** [comment]
> My first ruby - so any code tips and comments super welcome.

Oh cool, I'll go through and do that too now. My first general comment would be that Ruby indent style is 2 spaces instead of tabs.

File: scraper.rb [hunk 3]
```diff
@@ -1,3 +1,55 @@
-require File.join(File.dirname(__FILE__), "planningalerts_xml_feed", "planningalerts_xml_feed")
+require 'scraperwiki'
+require 'mechanize'
+
+agent = Mechanize.new
+
+def extract_from_p(contents, name)
+	contents.at('p:contains("'+name+'")').inner_text.split(name).last.strip
```
Comment (henare) [inline scraper.rb]: `+` usually has a space around it so the line would be:

```
contents.at('p:contains("' + name + '")').inner_text.split(name).last.strip
```

You could also use string interpolation to make it more readable but having to escape the double quotes negates that so the above is probably best.

File: scraper.rb [hunk 4]
```diff
@@ -1,3 +1,55 @@
...
+
+#get links to new developments
+
+apps_links = []
+
+links = page.search('a')
+links.each do |link|
+	href = link['href']
+
+	if (href and href.start_with?("http://www.salisbury.sa.gov.au/Build/Planning_Building_and_Forms/Advertised_Development_Applications/") )
```
Comment (henare) [inline scraper.rb]: Avoid the `and` keyword and instead use `&&`: https://github.com/bbatsov/ruby-style-guide#no-and-or-or

File: scraper.rb [hunk 5]
```diff
@@ -1,3 +1,55 @@
...
+		apps_links << href
+	end
+end
+
+apps_links.each_with_index do |app_url,index|
+	puts "Scraping application #{index} of #{apps_links.length} ..."
+	app = agent.get(app_url)
+	scrape_page(app);
```
Comment (henare) [inline scraper.rb]: You couldn't resist one semicolon, could you? :wink:

**Comment by henare** [comment]
> My first ruby - so any code tips and comments super welcome.

I've added a few comments but they're mostly style and very minor - great job :+1:

**Comment by e-e-e** [comment]
fixed.

**Comment by e-e-e** [comment]
awe dang semicolons are my default.

**Comment by e-e-e** [comment]
thanks for all the veteran pointers. I always default to tabs too, just for aesthetics in my text editor - lines begin with — rather than ••. Easy fix though.

The && vs and is interesting - inverse of python.

**Comment by henare** [comment]
Wee, great stuff @e-e-e https://www.planningalerts.org.au/authorities/salisbury/applications

The 132,499 people of Salisbury thank you :pray:

**Comment by e-e-e** [comment]
<3

**Pull request merged**